src/app/home-page/widget/entry-box/entry-box.component.ts

```diff
@@ -91,10 +92,11 @@ export class EntryBoxComponent extends Base implements OnInit {
       )
       .subscribe(
         (myEntries: Array<EntryUpdateTime>) => {
-          myEntries.forEach(() => {
-            this.listOfEntries = myEntries.slice(0, 7);
+          this.listOfEntries = myEntries.slice(0, 7);
+          if (this.firstCall) {
```

Comment:
Because the requests are asynchronous, there's no guarantee you'll get the response from the first request first. It's extremely unlikely in this scenario, given the likely lag between the page loading and a character being typed, but still possible. I know there was code like this before, but in hindsight it wasn't quite right.

Example:

1. Page loads, request to get entries without filter starts
2. Network is or becomes super slow, on cellular
3. User types in a filter string, a request to get (filtered) entries starts
4. Network switched over to high speed wifi, response for 3 comes back before response for 1.

One way to solve it would be to get the HttpResponse, like [here](https://github.com/dockstore/dockstore-ui2/blob/develop/src/app/workflows/list/published-workflows.datasource.ts#L61), then check to see if the filterText was set in the request. There might be other ways.